lib/chartmogul/api_resource.rb

```diff
@@ -22,6 +24,7 @@ def self.connection
       @connection ||= Faraday.new(url: ChartMogul::API_BASE) do |faraday|
         faraday.use Faraday::Request::BasicAuthentication, ChartMogul.account_token, ChartMogul.secret_key
         faraday.use Faraday::Response::RaiseError
+        faraday.request :retry, max: ChartMogul.max_retries, retry_statuses: RETRY_STATUSES
```

Comment:
To match the same behaviour with other clients, I think you need to use max_interval: 60 (1  minute), backoff_factor: 2 (for exponential backoff), interval_randomness: 0.5 (random jitter),  interval: 1 (initial delay). Also setting max_retries=0 doesn't seem to disable backoff, it defaults to 2 retries: https://github.com/lostisland/faraday/blob/0aebd35d9ff334e405f39ddcf2919388afd53bc1/lib/faraday/request/retry.rb#L40

Reply:
Also please check the exception option: https://github.com/lostisland/faraday/blob/0aebd35d9ff334e405f39ddcf2919388afd53bc1/lib/faraday/request/retry.rb#L94 You can use it to retry on network exceptions.